### PR TITLE
GATE-2293 added teams logging, fips and url browser isolation settings

### PR DESCRIPTION
**Pull request by rexscaria**
This PR adds logging + fips + browser isolation settings to team's account .

### CODE REVIEW

File: website/docs/r/teams_account.html.markdown [hunk 1]
```diff
@@ -34,19 +60,34 @@ The following arguments are supported:
 * `tls_decrypt_enabled` - (Optional) Indicator that decryption of TLS traffic is enabled.
 * `block_page` - (Optional) Configuration for a custom block page.
 * `antivirus` - (Optional) Configuration for antivirus traffic scanning.
+* `url_browser_isolation_enabled` - (Optional) Safely browse websites in Browser Isolation through a URL
+* `filps_tls` - (Optional) Enable only cipher suites and TLS versions compliant with FIPS 140-2.
```
Comment (shellcromancer) [inline website/docs/r/teams_account.html.markdown:64]:
```suggestion
* `fips_tls` - (Optional) Enable only cipher suites and TLS versions compliant with FIPS 140-2.
```
Remove `l`

File: cloudflare/resource_cloudflare_teams_accounts.go [hunk 2]
```diff
@@ -86,12 +120,34 @@ func resourceCloudflareTeamsAccountUpdate(d *schema.ResourceData, meta interface
 	if ok {
 		updatedTeamsAccount.Settings.ActivityLog = &cloudflare.TeamsActivityLog{Enabled: activtyLog.(bool)}
 	}
+
+	browserIsolation, ok := d.GetOkExists("url_browser_isolation_enabled")
+	if ok {
+		updatedTeamsAccount.Settings.BrowserIsolation = &cloudflare.BrowserIsolation{UrlBrowserIsolationEnabled: browserIsolation.(bool)}
+	}
+
+	fipsTls, ok := d.GetOkExists("fips_tls")
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_teams_accounts.go:129]: i think #1380 already covered this. you might need to rebase it in with your changes.

Comment (Michael9127) [reply cloudflare/resource_cloudflare_teams_accounts.go]: done

File: cloudflare/resource_cloudflare_teams_accounts_test.go [hunk 3]
```diff
@@ -40,7 +40,17 @@ func TestAccCloudflareTeamsAccountConfigurationBasic(t *testing.T) {
...
+					resource.TestCheckResourceAttr(name, "logging.0.settings_by_rule_type.0.dns.0.log_all", "false"),
+					resource.TestCheckResourceAttr(name, "logging.0.settings_by_rule_type.0.dns.0.log_blocks", "true"),
+					resource.TestCheckResourceAttr(name, "logging.0.settings_by_rule_type.0.http.0.log_all", "true"),
+					resource.TestCheckResourceAttr(name, "logging.0.settings_by_rule_type.0.http.0.log_blocks", "true"),
+					resource.TestCheckResourceAttr(name, "logging.0.settings_by_rule_type.0.l4.0.log_all", "false"),
+					resource.TestCheckResourceAttr(name, "logging.0.settings_by_rule_type.0.l4.0.log_blocks", "true"),
+					resource.TestCheckResourceAttr(name, "proxy.0.tcp", "true"),
+					resource.TestCheckResourceAttr(name, "proxy.0.udp", "false"),
 				),
+				ExpectNonEmptyPlan: true,
```
Comment (jacobbednarz) [inline cloudflare/resource_cloudflare_teams_accounts_test.go:53]: @rexscaria what is the expectation here? having a non-empty plan implies a permadiff or something isn't matching up.

File: website/docs/r/teams_account.html.markdown [hunk 4]
```diff
@@ -2,8 +2,7 @@
 layout: "cloudflare"
 page_title: "Cloudflare: cloudflare_teams_account"
 sidebar_current: "docs-cloudflare-resource-teams-account"
-description: |-
-Provides a Cloudflare Teams Account resource.
+description: |- Provides a Cloudflare Teams Account resource.
```
Comment (jacobbednarz) [inline website/docs/r/teams_account.html.markdown:5]:
```suggestion
description: Provides a Cloudflare Teams Account resource.
```

**Comment by jacobbednarz** [comment]
running the acceptance tests locally looks to be problematic too

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareTeamsAccount" -count 1 -parallel 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareTeamsAccountConfigurationBasic
    resource_cloudflare_teams_accounts_test.go:25: Step 1/1 error: Error running apply: exit status 1

        Error: error updating Teams Account configuration for account "f037e56e89293a057740de681ac9abbe": HTTP status 400: EOF

          with cloudflare_teams_account.cooxexhifi,
          on terraform_plugin_test.tf line 2, in resource "cloudflare_teams_account" "cooxexhifi":
           2: resource "cloudflare_teams_account" "cooxexhifi" {

--- FAIL: TestAccCloudflareTeamsAccountConfigurationBasic (10.10s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/cloudflare	10.692s
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
```

**Comment by jacobbednarz** [comment]
you might want to co-ordinate with @jgpaiva here too as it looks like you are both touching the same file and tests.

**Comment by jgpaiva** [comment]
@jacobbednarz I don't think so. 
Rex is modifying:
image
and I'm modifying:
image

i.e. Rex is `teams_accounts` and I'm `teams_rules` 😃

File: website/docs/r/teams_account.html.markdown [hunk 5]
```diff
@@ -35,8 +65,10 @@ The following arguments are supported:
 * `block_page` - (Optional) Configuration for a custom block page.
 * `fips` - (Optional) Configure compliance with Federal Information Processing Standards
 * `antivirus` - (Optional) Configuration for antivirus traffic scanning.
+* `url_browser_isolation_enabled` - (Optional) Safely browse websites in Browser Isolation through a URL
```
Comment (jacobbednarz) [inline website/docs/r/teams_account.html.markdown:68]:
```suggestion
* `url_browser_isolation_enabled` - (Optional) Safely browse websites in Browser Isolation through a URL.
```

File: website/docs/r/teams_account.html.markdown [hunk 6]
```diff
@@ -47,10 +79,27 @@ The **block_page** block supports:
 The **FIPS** block supports:
 * `tls` - (Optional) Only allow FIPS-compliant TLS configuration
 
+The **antivirus** block supports:
+
+* `enabled_download_phase` - (Optional) Scan on file download.
+* `enabled_upload_phase` - (Optional) San on file upload.
```
Comment (jacobbednarz) [inline website/docs/r/teams_account.html.markdown:85]:
```suggestion
* `enabled_upload_phase` - (Optional) Scan on file upload.
```

File: cloudflare/resource_cloudflare_teams_accounts_test.go (hunk 3 above)
Comment (jacobbednarz) [reply cloudflare/resource_cloudflare_teams_accounts_test.go:53]: if it is this output, this is a current bug that needs to be fixed from another PR. we can merge without this and just fix it in another PR.

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareTeamsAccount" -count 1 -parallel 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareTeamsAccountConfigurationBasic
    resource_cloudflare_teams_accounts_test.go:25: Step 1/1 error: After applying this test step, the plan was not empty.
        stdout:


        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # cloudflare_teams_account.qgmqrtaguz will be updated in-place
          ~ resource "cloudflare_teams_account" "qgmqrtaguz" {
              - activity_log_enabled = true -> null
                id                   = "f037e56e89293a057740de681ac9abbe"
                # (2 unchanged attributes hidden)





                # (5 unchanged blocks hidden)
            }

        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccCloudflareTeamsAccountConfigurationBasic (9.03s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/cloudflare	9.578s
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
FAIL
make: *** [testacc] Error 1
```

Comment (jacobbednarz) [reply cloudflare/resource_cloudflare_teams_accounts_test.go:53]: fixed

**Comment by jacobbednarz** [comment]
acceptance tests are passing

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareTeamsAccount" -count 1 -parallel 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareTeamsAccountConfigurationBasic
--- PASS: TestAccCloudflareTeamsAccountConfigurationBasic (11.79s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/cloudflare	12.036s
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
```

**Pull request merged**